### ISSUE TITLE
Static renderers ignore height/width

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.py#2651]
This always renders at the default size for me:

```python
import plotly.express as px
fig = px.scatter(y=[1,2,3], height=300, width=300)
fig.show("png")
```

I would consider this a bug, although there may be some reasoning behind this behaviour ... @jonmmease ?

**Comment by jonmmease** [comment]
Yeah, this behavior is not intuitive.  The issue is that the `height` and `width` here are overriding the `height` and `width` properties of the Image renderer.  So this is roughly equivalend to 
```
import plotly.io as pio
pio.renderers["png"].height = 300
pio.renderers["png"].width = 300

import plotly.express as px
fig = px.scatter(y=[1,2,3])
fig.show("png")
```

So here, `width` and `height` are really the default width and height. And here I think it makes sense for the figure's own width/height to take precedence.

But in any case, we do need to put some more thought into precedence.

**Comment by nicolaskruchten** [comment]
Right so this is the same as `fig.show("png", width=300, height=300)`

**Comment by nicolaskruchten** [comment]
See also #2653

**Comment by nicolaskruchten** [comment]
This is addressed in #3240 and will be part of version 5.0
